drivers/dma/dma_mcux_edma.c

```diff
@@ -24,17 +24,37 @@
 #include <zephyr/logging/log.h>
 #include <zephyr/irq.h>
 
+#ifdef CONFIG_DMA_MCUX_EDMA_LEGACY
 #define DT_DRV_COMPAT nxp_mcux_edma
+#elif CONFIG_DMA_MCUX_EDMA_V3
+#define DT_DRV_COMPAT nxp_mcux_edma_v3
+#endif
 
 LOG_MODULE_REGISTER(dma_mcux_edma, CONFIG_DMA_LOG_LEVEL);
 
+#define HAS_CHANNEL_GAP(inst) DT_INST_NODE_HAS_PROP(inst, channel_gap) ||
+
+#define DMA_MCUX_HAS_CHANNEL_GAP (DT_INST_FOREACH_STATUS_OKAY(HAS_CHANNEL_GAP) 0)
+
 struct dma_mcux_edma_config {
 	DMA_Type *base;
-	DMAMUX_Type *dmamux_base;
+	DMAMUX_Type **dmamux_base;
```

Comment:
for rt11xx and before although we have multiply dma mux, but they belongs to different domain, mean they can not be access at the same time. but for S32K, one DMA can use multiply dmamux, this is different.

Reply:
Yep, i understand your point. The dmamux is declared here is the one that can route DMA sources for the controller, not all dmamuxs available in the SoC. Perhaps the `reg-names` cause confusion, i've updated